### PR TITLE
Ruby: parenthesized patterns

### PR DESCRIPTION
**Pull request by aibaars**
This pull request updates the grammar and adds a class for `ParenthesizedPattern`s.

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/ast/Pattern.qll [hunk 1]
```diff
@@ -389,6 +389,30 @@ class AsPattern extends CasePattern, TAsPattern {
   }
 }
 
+/**
+ * A parenthesized pattern:
+ * ```rb
+ * in (1 ..)
+ * in (0 | "" | [] | {})
+ */
```
Comment (nickrolfe) [inline ruby/ql/lib/codeql/ruby/ast/Pattern.qll:397]: Missing backticks.

**Review by nickrolfe — COMMENTED** [review]
Otherwise, LGTM.

File: ruby/ql/lib/codeql/ruby/ast/Pattern.qll (hunk 1 above)
Comment (aibaars) [inline ruby/ql/lib/codeql/ruby/ast/Pattern.qll:397]:
```suggestion
 * ```
 */
```

Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/Pattern.qll:398]: done!

**Review by nickrolfe — APPROVED** [review]

**Pull request merged**